### PR TITLE
[consensus] better error handling of execution failure

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

This commit extends the notify to mempool to support failed block
execution.

This can help address issues when a txn successfully executed but causes
internal error in the execution e.g. rotate to invalid validator network addresses.

Before this commit, such a txn can cause all exeuction fail until the
txn is expired in mempool.

After this commit, such a txn should be dropped immediately after the
first failure.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

will add one when we have real types back in ValidatorConfig, currently it can't fail

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/src/state_replication.rs [hunk 1]
```diff
@@ -17,7 +17,11 @@ pub trait TxnManager: Send + Sync {
 
     /// Notifies TxnManager about the executed result of the block,
     /// which includes the specifics of what transactions succeeded and failed.
-    async fn notify(&self, block: &Block, compute_result: &StateComputeResult) -> Result<()>;
+    async fn notify(
+        &self,
+        block: &Block,
+        executed_result: Result<&StateComputeResult>,
```
Comment (davidiw) [inline consensus/src/state_replication.rs:23]: why don't we embed the result in the StateComputeResult? this feels a little redundant... will read more ... but figured easier to ask a naive question.

Comment (davidiw) [reply consensus/src/state_replication.rs:23]: Hmm.... @zekun000

Comment (zekun000) [reply consensus/src/state_replication.rs:23]: StateComputeResult represents the successful executed result, I don't understand how do we embed the Result into that? that means we need another struct StateComputeSuccessfulResult to represent the existing struct, don't see what's the benefit.

**Comment by zekun000** [comment]
turns out it's not possible for this to happen after we change network address to Vec<u8> in ValidatorConfig, closing

**Pull request closed without merging**